### PR TITLE
[ONNX] Add an ATen fallback pathway for ONNX export

### PR DESCRIPTION
**Pull request by jamesr66a**
This adds a flag that makes it so that if we cannot export an operator, we will fall back to exporting an op of type ATen with the appropriate schema. This will help smooth over our production use cases, where if ONNX supports and op we want to use that translation pathway, but if it does not we want to simply use ATen for speed of deployment

### CODE REVIEW

File: torch/csrc/jit/export.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@ std::string value_name(Value* n) {
 struct ExportContext {
   size_t num_blocks = 0;
   bool export_raw_ir = false;
+  bool aten_fallback = false;
```
Comment (dzhulgakov) [inline torch/csrc/jit/export.cpp]: time to make it an enum - too many bool fields flying around.

Basically we have:
- raw
- onnx strict
- onnx with Aten ops everywhere (added by Lu)
- onnx with Aten ops as fallback

I think export mode would make it much saner - with your diff we now have 3 bool fields :( And it's pretty confusing for this code maintainability

**Review by dzhulgakov — CHANGES_REQUESTED** [review]

File: torch/csrc/Module.cpp [hunk 2]
```diff
@@ -517,8 +517,8 @@ static PyObject* initModule() {
   ASSERT_TRUE(THPVariable_initModule(module));
   ASSERT_TRUE(THPFunction_initModule(module));
   ASSERT_TRUE(THPEngine_initModule(module));
-  torch::jit::initJITBindings(module);
   torch::onnx::initONNXBindings(module);
+  torch::jit::initJITBindings(module);
```
Comment (dzhulgakov) [inline torch/csrc/Module.cpp]: please add comments that the order is important to preserve future pain :)

**Review by dzhulgakov — APPROVED** [review]
Looks great, thanks for making code to have less unnamed bools :)

File: torch/onnx/utils.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def set_training(model, mode):
 
 
 def export(model, args, f, export_params=True, verbose=False, training=False,
-           input_names=None, output_names=None, aten=False, export_raw_ir=False):
+           input_names=None, output_names=None, operator_export_type=OperatorExportTypes.ONNX):
```
Comment (apaszke) [inline torch/onnx/utils.py]: This changes the public interface? Did we advertise the bool args previously? If yes, add bc compat code to translate them to enum values

File: torch/onnx/utils.py [hunk 4]
```diff
@@ -403,7 +404,9 @@ def _run_symbolic_function(g, n, inputs, env, aten=False):
             return None
 
         elif ns == "aten":
-            if aten:
+            is_exportable_aten_op = hasattr(torch.onnx.symbolic, op_name)
+            if (operator_export_type == OperatorExportTypes.ONNX_ATEN) or \
+                    (not is_exportable_aten_op and operator_export_type == OperatorExportTypes.ONNX_ATEN_FALLBACK):
```
Comment (apaszke) [inline torch/onnx/utils.py]: Can you please split the two branches in this if into variables and use them in here? It's very hard to read at the moment.

**Review by apaszke — APPROVED** [review]

**Pull request merged**